app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -133,4 +133,30 @@ private void showFilterMenu(View view) {
         popupMenu.show();
     }
 
-}
+    /**
+     * Called when a filter is applied or modified.
+     * @param filterType  The type of filter being applied (e.g., "MAKE").
+     * @param filterValue The value or criteria for the filter.
+     */
+    @Override
+    public void onFilterApplied(String filterType, String filterValue) {
+        appliedFilters.put(filterType, filterValue);
+        ArrayList<Item> filteredList = new ArrayList<>();
+
+        for (Map.Entry<String, Object> filter : appliedFilters.entrySet()) {
+            switch (filter.getKey()) {
+                case "MAKE":
```

Comment:
This works, and I don't think we have to change it, however maybe we could make appliedFilters a set of filter objects and then call a .filter() method instead that takes ArrayList<Item> as an input and returns the filtered ArrayList. @ocooke Thoughts?

Reply:
Yeah I think that's a good idea. Especially because we want to let them apply multiple filters simultaneously. So we don't want separate case statements for each. 

There's also the argument that we may want the HomePage to reference the "filtered" list at all times. As in:
Firebase collection -> mapped to ArrayList -> applies current Filters -> filtered list is the list that is displayed on HomePage.

Reply:
I've tried to address this design issue. Please see the updated PR.